### PR TITLE
Two CLI options for oauth-enabled installations

### PR DESCRIPTION
**Pull request by vrutkovs**
Includes #375

 * print-token-url: print an auth page url to get token
 * get-serviceaccount-token: prints tokens set for serviceaccount

### CODE REVIEW

File: osbs/cli/main.py [hunk 1]
```diff
@@ -349,6 +356,17 @@ def cmd_restore(args, osbs):
     logger.info("backup recovery complete!")
 
 
+def cmd_oauth_token(args, osbs):
+    uri = urljoin(osbs.os_conf.get_openshift_base_uri(), "/token/request")
```
Comment (twaugh) [inline osbs/cli/main.py]: Should this be `/oauth/token/request`?

Comment (vrutkovs) [inline osbs/cli/main.py]: Indeed, gonna fix that, although my test instance somehow managed to redirect correctly

File: osbs/cli/main.py [hunk 2]
```diff
@@ -499,6 +517,17 @@ def cli():
                                  help="don't stop when restoring a resource fails")
     restore_builder.set_defaults(func=cmd_restore)
 
+    oauth2_builder = subparsers.add_parser(str_on_2_unicode_on_3('print-oauth-token-url'),
```
Comment (twaugh) [inline osbs/cli/main.py]: We ought to have some consistency about how we refer to tokens. Here it's 'oauth-token', elsewhere it's 'oauth2-token' and 'token'.

Should we just use 'token' throughout?  e.g. osbs --token ..., "token = " in osbs.conf, --print-token-url, get-serviceaccount-token (as now).

Comment (vrutkovs) [inline osbs/cli/main.py]: Okay, lets use 'token' (the auth page also uses that)

**Comment by twaugh** [comment]
Looks really nice.

I'd like to use this PR to also figure out a way of storing the service account token separately from the config file. Maybe a 'token_file' configuration parameter?

Then I think 'get-serviceaccount-token' could output just the token on its own without the secret's name, so we could do this:

```
osbs --token $ADMIN_USER_TOKEN get-serviceaccount-token $SERVICEACCOUNT > sa-token
osbs --token-file sa-token ...
```

**Comment by vrutkovs** [comment]
get-serviceaccount-token may return several tokens, a simple
redirection will not work in some cases.

What if we add '--save-in-config' to overwrite osbs.conf param with
first found token?

**Comment by twaugh** [comment]
OK, so perhaps let's just implement a `--token-file` argument/config param, so the service account token can be stored outside of osbs.conf.

**Comment by vrutkovs** [comment]
Done, now we can read token from file too.

Reverted 'deprecate ```use_auth``` and look for ```use_httpd_auth```' change - its still required for oauth to work

**Comment by twaugh** [comment]
OK, figured out what wasn't working for me: we need to base64-decode the data in the token secret before displaying it.

**Comment by twaugh** [comment]
LGTM

**Comment by twaugh** [comment]
It needs docs, but I'll file a separate issue for that.

**Pull request merged**